core/src/main/java/org/elasticsearch/index/query/HasParentQueryParser.java

```diff
@@ -42,16 +43,12 @@
 import java.util.Set;
 
 
-public class HasParentQueryParser extends BaseQueryParserTemp {
+public class HasParentQueryParser extends BaseQueryParser  {
 
+    private static final HasParentQueryBuilder PROTOTYPE = new HasParentQueryBuilder("", EmptyQueryBuilder.PROTOTYPE);
```

Comment:
makes sense, was about to say we could move all existing prototypes to parsers, but if we then we want to merge parser and builder, let's not do it till we made a decision there ;)